[CodeCompletion] Don’t compute isolated parameter captures during code completion

ahoppen opened the pull request:
Computing capture information requires a type checked AST, which we don’t have during code completion. To fix an assertion failure, don’t look for a captured `isolated` parameter while computing a closure’s actor isolation during code completion.

rdar://126923558

ahoppen commented:
@swift-ci Please smoke test

hamishknight commented on lib/Sema/TypeCheckConcurrency.cpp:
```diff
@@ -4167,9 +4173,16 @@ namespace {
-        if (auto param = closure->getCaptureInfo().getIsolatedParamCapture())
-          return ActorIsolation::forActorInstanceCapture(param)
-            .withPreconcurrency(preconcurrency);
+        if (checkIsolatedCapture) {
+          if (auto param = closure->getCaptureInfo().getIsolatedParamCapture())
+            return ActorIsolation::forActorInstanceCapture(param)
+                .withPreconcurrency(preconcurrency);
+        } else {
+          // If we don't have capture information during code completion, assume
+          // that the closure captures the `isolated` parameter from the parent
+          // context.
+          return parentIsolation;
+        }
```
I guess an alternative solution would be to call `computeCaptures` for completion to just populate an empty set of captures. Can't say I really love either solution, but I think I would mildly prefer `computeCaptures` since it's less invasive.

hamishknight commented on validation-test/IDE/crashers_2_fixed/rdar126923558.swift:
```diff
@@ -0,0 +1,9 @@
+  func bar() {}
+}
+
+func test(foo: isolated Foo) {
+  foo.#^COMPLETE^#
+}
```
This test case doesn't have any closures, does it trigger the crash without the fix?

ahoppen replied:
Oh, yes. Indeed. I was trying other things in there and then forgot to reset it. Thanks for noticing.

ahoppen committed:
[CodeCompletion] Don’t compute isolated parameter captures during code completion

Computing capture information requires a type checked AST, which we don’t have during code completion. To fix an assertion failure, don’t look for a captured `isolated` parameter while computing a closure’s actor isolation during code completion.

rdar://126923558

ahoppen commented:
@swift-ci Please smoke test
